### PR TITLE
Implement floating-point manipulation functions for `BigFloat`

### PR DESCRIPTION
**Pull request by HertzDevil**
`BigFloat` already has an implementation of `frexp` because `hash` needs it (I think); this PR adds the remaining floating-point manipulation operations.

Methods that take an additional integer accept an `Int` directly, so this takes care of the allowed conversions in #10907.

`BigFloat` from GMP doesn't have the notion of signed zeros, so `copysign` is only an approximation. (As usual, MPFR floats feature signed zeros.)

### CODE REVIEW

File: src/big/big_float.cr [hunk 1]
```diff
@@ -393,6 +393,43 @@ class String
...
+    LibGMP.mpf_get_d_2exp(out exp, value)
+    (exp - 1).to_i64
+  end
+
+  # Returns the unbiased radix-independent exponent of the given floating-point *value*.
+  #
+  # For `BigFloat` this is equivalent to `ilogb`.
+  def logb(value : BigFloat) : BigFloat
+    LibGMP.mpf_get_d_2exp(out exp, value)
+    (exp - 1).to_big_f
```
Comment (straight-shoota) [inline src/big/big_float.cr:407]: Why not just make this call the other method when it's equivalent?
```suggestion
    ilogb(value)
```

Comment (HertzDevil) [reply src/big/big_float.cr:407]: Their return types are different (I don't know why C does the same, to be honest).

Comment (HertzDevil) [reply src/big/big_float.cr:407]: If `(exp - 1).to_i64.to_big_f` never leads to a loss of information then `ilogb(value).to_big_f` is okay.

**Comment by straight-shoota** [comment]
CI is failing on Windows: https://github.com/crystal-lang/crystal/actions/runs/10698882947/job/29660333970?pr=11007

```
  1) BigFloat Math .frexp
     Failure/Error: Math.frexp(2.to_big_f ** 1_000_000_000_000).should eq({0.5, 1_000_000_000_001})

       Expected: {0.5, 1000000000001}
            got: {2.43829426113685079336e+327749382, -727379967}

     # spec\std\big\big_float_spec.cr:585

  2) BigFloat Math .logb
     Failure/Error: Math.ilogb(2.to_big_f ** 1_000_000_000_000).should eq(1_000_000_000_000.to_big_f)

       Expected: 1000000000000.0
            got: -727379968

     # spec\std\big\big_float_spec.cr:559

  3) BigFloat Math .ilogb
     Failure/Error: Math.ilogb(2.to_big_f ** 1_000_000_000_000).should eq(1_000_000_000_000)

       Expected: 1000000000000
            got: -727379968

     # spec\std\big\big_float_spec.cr:553
```

**Comment by HertzDevil** [comment]
These exponents will overflow even when GMP / MPIR handles the internal exponents correctly, so a fix similar to #14971 is needed

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**